tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.html

```diff
@@ -0,0 +1,284 @@
+    import {runsColorScale} from "../tf-color-scale/colorScale.js";
+    import * as ChartHelpers from '../vz-line-chart/vz-chart-helpers.js';
+
+    Polymer({
+      is: "tf-pr-curve-card",
+      properties: {
+        runs: Array,
+        tag: String,
+        // For each run, the card will display the PR curve at this step or the one
+        // closest to it, but less than it.
+        stepCapPerRun: Object,
+        // This maps a run to the actual step that is presented in the PR curve.
+        _actualStepPerRun: {
+          type: Object,
+          value: {},
+        },
+        // This maps a run to the actual wall time that is presented in the PR
+        // curve.
+        _actualWallTimePerRun: {
+          type: Object,
+          value: {},
+        },
+        // A list of runs with an available step. Used to populate the table of
+        // steps per run.
+        _runsWithStepAvailable: {
+          type: Array,
+          computed: "_computeRunsWithStepAvailable(_actualStepPerRun)",
+        },
+        _runToData: Object,
+        /** @type {Function} */
+        _colorScaleFunction: {
+          type: Object,  // function: string => string
+          value: () => ({scale: runsColorScale}),
+        },
+        requestManager: Object,
+        _canceller: {
+          type: Object,
+          value: () => new Canceller(),
+        },
+        _attached: Boolean,
+        // The value field is a function that returns a function because Polymer
+        // will actually call the value field if the field is a function. However,
+        // we actually want the value itself to be a function.
+        _xComponentsCreationMethod: {
+          type: Object,
+          value: () => (() => {
+            const scale = new Plottable.Scales.Linear();
+            return {
+              scale: scale,
+              axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+              accessor: d => d.recall,
+            };
+          }),
+          readOnly: true,
+        },
+      },
+      observers: [
+          'reload(run, tag)',
+          '_runsChanged(_attached, runs.*)',
+          '_setChartData(_runToData, stepCapPerRun)',
+      ],
+      _computeRunColor(run) {
+        return this._colorScaleFunction.scale(run);
+      },
+      attached() {
+        // Defer reloading until after we're attached, because that ensures that
+        // the requestManager has been set from above. (Polymer is tricky
+        // sometimes)
+        this._attached = true;
+        this.reload();
+      },
+      reload() {
+        if (!this._attached) {
+          return;
+        }
+        this._canceller.cancelAll();
+        const router = getRouter();
+        let url = router.pluginRoute('pr_curves', '/pr_curves');
+        url += url.indexOf('?') > -1 ? '&' : '?';
+        const urlRunsPortion = this.runs.map(r => `run=${r}`).join('&');
+        url += `tag=${this.tag}&${urlRunsPortion}`;
```

Comment:
nit: Better to let ``const urlRunsPortion = this.runs.map(r => `&run=${r}`).join('');`` and ``url += `tag=${this.tag}${urlRunsPortion}`;`` so that when there are no runs we don't have a trailing ampersand.